tools/install-powershell.ps1

```diff
@@ -85,25 +107,75 @@ try {
         $metadata = Invoke-RestMethod https://api.github.com/repos/powershell/powershell/releases/latest
         $release = $metadata.tag_name -replace '^v'
 
-        $packageName = "PowerShell-${release}-win-${architecture}.zip"
+        $packageName = if ($IsWinEnv) {
+            "PowerShell-${release}-win-${architecture}.zip"
+        } elseif ($IsLinuxEnv) {
+            "powershell-${release}-linux-${architecture}.tar.gz"
+        } elseif ($IsMacOSEnv) {
+            "powershell-${release}-osx-${architecture}.tar.gz"
+        }
+
         $downloadURL = "https://github.com/PowerShell/PowerShell/releases/download/v${release}/${packageName}"
         Write-Verbose "About to download package from '$downloadURL'" -Verbose
 
         $packagePath = Join-Path -Path $tempDir -ChildPath $packageName
         Invoke-WebRequest -Uri $downloadURL -OutFile $packagePath
 
-        Expand-Archive -Path $packagePath -DestinationPath $Destination
+        if ($IsWinEnv) {
+            Expand-Archive -Path $packagePath -DestinationPath $Destination
+        } else {
+            tar zxf $packagePath -C $Destination
+        }
     }
 
-    if ($AddToPath -and (-not $env:Path.Contains($Destination))) {
-        ## Add to the User scope 'Path' environment variable
-        $userPath = [System.Environment]::GetEnvironmentVariable("Path", "User")
-        $userPath = $Destination + [System.IO.Path]::PathSeparator + $userPath
-        [System.Environment]::SetEnvironmentVariable("Path", $userPath, "User")
+    ## Change the mode of 'pwsh' to 'rwxr-xr-x' to allow execution
+    if (-not $IsWinEnv) { chmod 755 "$Destination/pwsh" }
 
-        ## Add to the 'Path' for the current process
-        $env:Path = $Destination + [System.IO.Path]::PathSeparator + $env:Path
-        Write-Verbose "'$Destination' is added to Path" -Verbose
+    if ($AddToPath) {
+        if ($IsWinEnv -and (-not $env:Path.Contains($Destination))) {
+            ## Add to the User scope 'Path' environment variable
+            $userPath = [System.Environment]::GetEnvironmentVariable("Path", "User")
+            $userPath = $Destination + [System.IO.Path]::PathSeparator + $userPath
+            [System.Environment]::SetEnvironmentVariable("Path", $userPath, "User")
+            Write-Verbose "'$Destination' is added to the Path" -Verbose
```

Comment:
I think, since we have CmdletBinding() we should not force -verbose

Reply:
These messages are actually intentionally written out so that a user roughly knows what step the script is doing.